File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

// ask the user for an input between 1 and 23, print a double half pyramid of specified height
int main(void)
{
    int height;
    // ensure input is between 1 and 23
    do
    {
        height = get_int("Height: ");
    } while (height < 0 || height > 23);
    // print out each row
    for (int i = 0; i < height; i++)
    {
        // print the left gap
        for (int j = height - i - 1; j > 0; j--)
        {
            printf(" ");
        }
        // print the left pyramid
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }
        // print the space between pyramids
        printf("  ");
        // print the right pyramid
        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: pset6/mario/mario.py
def get_positive_int():
    """Ask user for a positive integer between 0 and 23"""
    while True:
        try:
            height = int(input("Height: "), 10)
        except ValueError:
            continue
        if height >= 0 and height < 24:
            return height
        else:
            continue


def mario():
    """Print a double half pyramid of a specified height"""
    height = get_positive_int()
    for row in range(1, height + 1):
        for _space in range(height - row, 0, -1):
            print(" ", end="")
        for _brick in range(row):
            print("#", end="")
        print("  ", end="")
        for _brick in range(row):
            print("#", end="")
        print()


if __name__ == "__main__":
    mario()
